tensorflow/python/ops/script_ops.py

```diff
@@ -116,6 +119,10 @@ def _convert(self, value, dtype):
       # TODO(akshayka): Make it possible to return a list of both Tensors and
       # Nones from an EagerPyFunc.
       return constant_op.constant(0.0, dtype=dtype)
+    if sparse_tensor.is_sparse(value):
+      value = sparse_ops.sparse_tensor_to_dense(value)
```

Comment:
@Leslie-Fang This solves the error, but the feature request was about directly supporting the tf.SparseTensors
--> After calling tf.py_function with a SparseTensor as return value, the output should be SparseTensor, and not Dense Tensor converted from the SparseTensor.

Reply:
Would that solves the problem?

````
    if sparse_tensor.is_sparse(value) or ragged_tensor.is_ragged(value):
        result = value
    else:
        result = ops.convert_to_tensor(value, dtype=dtype)

    return result
````

Reply:
Hi @VitaliKaiser I believe ops.convert_to_tensor will also cast the the data type of each element into dtype.

Reply:
I have made several changes to return the sparse_tensor or the ragged_tensor but still get the error:
```
tensorflow.python.framework.errors_impl.InternalError: Unexpected PyObject was returned: SparseTensor [Op:EagerPyFunc]
```
Does py_func have to return the eager tensor?

Reply:
The EagerPyFunc has to return eager tensors; sparsetensor is not a tensor, it's a python type with 3 tensors under it. If you return the 3 underlying tensors things should be fine

Reply:
Thanks @alextp If we return the sparse_tensor as a list of 3 eager_tensors. Does the user needs convert the 3 eager_tensors back into the sparse_tensor? Do you have any suggestions?